### ISSUE TITLE
VlcControl.Stop() deadlock and freeze application!

### ISSUE THREAD

**Issue by lebtho1878** [ZeBobo5/Vlc.DotNet#542]
SourceProvider.MediaPlayer.Stop();

freezes the application

**Comment by jeremyVignelles** [comment]
Can you share your full code, please?

Did you try to call `Stop()` from a VLC callback? https://github.com/ZeBobo5/Vlc.DotNet/wiki/Vlc.DotNet-freezes-(don't-call-Vlc.DotNet-from-a-Vlc.DotNet-callback)

**Comment by lebtho1878** [comment]
This is in  your example Samples.Wpf.Advanced, I changed to Stop

    private void OnStopButtonClick(object sender, RoutedEventArgs e)
        {
            control.SourceProvider.MediaPlayer.Stop();
            //this.control?.Dispose();
            //this.control = null;
        }

**Comment by jeremyVignelles** [comment]
While I still don't know why the app freeze, I found a workaround:

```
            ThreadPool.QueueUserWorkItem(_ => control.SourceProvider.MediaPlayer.Stop());
```

**Comment by lebtho1878** [comment]
Workaround works for me. Thank you!

**Comment by wangning08115** [comment]
yes, If you use event function in  VLC Display Callback function， then app will freeze.

**Comment by jeremyVignelles** [comment]
Yes, but that's not the case here, Stop() is invoked on the main thread because it is triggered by a mouse click. I guess that the rule is : don't call Stop() on the same thread that created the Player, but I'm not sure why it does that...

**Comment by marksard** [comment]
For your information.
Don't call Stop(), Pause() and Play() from main thread.
If call then deadlock.

**Comment by AqlaSolutions** [comment]
I have freeze on Stop too and I'm not calling it from events. 
![изображение](https://user-images.githubusercontent.com/3465565/70486905-2ac73300-1afc-11ea-8d40-7ac90e2bb73c.png)

**Comment by sunky0** [comment]
@jeremyVignelles 
**EDIT : Just realized that it's in the WPF control. it must be because the Video callback wants to go to the main thread, locking the callback, while the main thread waits for the Stop() to complete.**

Thank you for information that the main thread waits for the Stop() to complete.
I have a question.
It's just stop() function? How about Play(), Pause(), Dispose()?
